### PR TITLE
fix(forms)!: rewrite row to container, drop 'row' from ComponentInitializedEvent

### PR DESCRIPTION
**Pull request by dereekb**
# Pull Request

## Description

The RowFieldComponent was removed/replaced with the addition of Wrappers, so now the row-type declaration is actually a container with a row wrapper attached.

There were a few missing tests and type changes related to this though. These fixes address those issues.

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] 🎨 Code style update (formatting, renaming)
- [ ] ♻️ Code refactoring (no functional changes)
- [ ] ⚡ Performance improvement
- [ ] ✅ Test update
- [ ] 🔧 Build/tooling update

## Related Issues

Fixes #
Related to #

## Changes Made

- Bug where the Row type still utilized 'row' intead of 'container' after the row mapper was applied. Since the RowField no longer exists and is not registered any field using 'row' would throw a warning instead.
- Containers can now formally be placed within other Containers. Since rows were now containers we had to test to make sure containers could belong within Containers. No logic changes were made here.
- Updated ContainerAllowedChildren so they can now contain other Containers. This was a typing change.
- RowAllowedChildren now extends ContainerAllowedChildren
- Container can now include HiddenField
- Since the 'row' type is a ContainerField in the end, removed the row-related warnings for Hidden fields.

## Testing

- [x] Unit tests added/updated
- [x] All tests pass locally
- [ ] Tested manually
- [ ] Tested edge cases

## Documentation

- [x] Documentation updated (if needed)
- [ ] API documentation (TSDoc) added/updated
- [ ] Examples updated (if needed)

## Checklist

- [x] Code follows the [coding standards](../CODING_STANDARDS.md)
- [x] No linting errors (`pnpm lint`)
- [x] Code is formatted (`pnpm lint:fix`)
- [x] Build succeeds (`pnpm build:libs`)
- [x] Commits follow [conventional commit format](https://www.conventionalcommits.org/)
- [x] Self-review completed
- [x] No commented-out code or console.log() statements
- [x] Type safety maintained (no `any` types)

### CODE REVIEW

File: packages/dynamic-forms/src/lib/models/types/nesting-constraints.ts [hunk 1]
```diff
@@ -39,8 +32,22 @@ export type GroupAllowedChildren = LeafFieldTypes | RowField | ContainerField;
...
- * Containers follow the same rules as rows: groups, arrays, and leaf fields
- * (but NOT pages, rows, or hidden fields)
+ * Fields that are allowed as children of Container fields.
+ * Containers are pure layout primitives that flatten their children into the
+ * parent form, so any registered field type may appear inside — including
+ * pages, hidden fields, and other containers.
+ */
+export type ContainerAllowedChildren =
+  | LeafFieldTypes
+  | PageField
```
Comment (antimprisacaru) [inline packages/dynamic-forms/src/lib/models/types/nesting-constraints.ts]: So unfortunately, the current general design doesn't actually permit this. The framework expects the pages to be at top level, the dynamic form component currently has a separate flow that routes these through an orchestrator component. Having a page under a container will for sure break things.

File: packages/dynamic-forms/src/lib/fields/container/container-field.component.ts [hunk 2]
```diff
@@ -108,18 +108,7 @@ export default class ContainerFieldComponent {
...
-    // here so `(events)` consumers filtering by `componentType === 'row'`
-    // keep receiving emissions.
-    setupContainerInitEffect(
-      this.resolvedFields,
-      this.eventBus,
-      () => ((this.field().type as string) === 'row' ? 'row' : 'container'),
-      () => this.field().key,
-      this.injector,
-    );
+    setupContainerInitEffect(this.resolvedFields, this.eventBus, 'container', () => this.field().key, this.injector);
```
Comment (antimprisacaru) [inline packages/dynamic-forms/src/lib/fields/container/container-field.component.ts:111]: I think we should also remove the row entry from `ComponentInitializedEvent.componentType`. This will be a breaking change.

File: packages/dynamic-forms/src/lib/fields/hidden/hidden-field.spec.ts [hunk 3]
```diff
@@ -113,68 +112,4 @@ describe('HiddenField', () => {
       expect(isHiddenField(field)).toBe(false);
     });
   });
-
-  describe('row nesting validation (warnings)', () => {
```
Comment (antimprisacaru) [inline packages/dynamic-forms/src/lib/fields/hidden/hidden-field.spec.ts:117]: Missing positive test for the new permissive behavior. The PR deletes the four row-nesting tests, but does not add a counterpart that asserts a row containing a hidden field now renders / submits / fires no warning. Without it, the previously-validated invariant is removed and nothing locks in the new one.

**Review by antimprisacaru — COMMENTED** [review]
Could you please also update the MCP to reflect the new constraints of field nesting?

- in `registry/field-types.ts:340,366,378` — row description still says "NOT ALLOWED: pages, rows, hidden fields" and cannotContain: ['page', 'row', 'hidden'].

- in `tools/data/lookup-topics.ts:356-358,1683,1694` — "NOT allowed: hidden fields, page containers, nested row containers" and "Common mistake: Putting hidden fields inside rows".

AI agents driving form generation will keep gate-keeping against patterns this PR enables.

File: packages/dynamic-forms/src/lib/models/types/nesting-constraints.ts (hunk 1 above)
Comment (dereekb) [reply packages/dynamic-forms/src/lib/models/types/nesting-constraints.ts]: Ah, good catch. PageField shouldn't be there.

File: packages/dynamic-forms/src/lib/fields/container/container-field.component.ts (hunk 2 above)
Comment (dereekb) [reply packages/dynamic-forms/src/lib/fields/container/container-field.component.ts:111]: Done

File: packages/dynamic-forms/src/lib/fields/hidden/hidden-field.spec.ts (hunk 3 above)
Comment (dereekb) [reply packages/dynamic-forms/src/lib/fields/hidden/hidden-field.spec.ts:117]: Done

**Comment by dereekb** [comment]
Ok, all comments should have been addressed now

**Comment by antimprisacaru** [comment]
All good from my side @dereekb, fix the CI and I'll approve. Might be tomorrow though because it's midnight here 😅

**Comment by dereekb** [comment]
Found one more bug where a container with a nested container in an ArrayField didn't emit the correct value. I was getting a value like:

```json
> {
  "addresses": [
    {
      "line1": "",
      "line2": "",
      "city": "",
      "_row_49": {
        "state": "",
        "zip": ""
      },
      "country": ""
    }
  ]
}
```

Where `_row_49` was a row field. Fixed now in the latest commit.

**Review by antimprisacaru — COMMENTED** [review]
Nicely done, thanks for the contribution! And nice catch with that bug!
